projects/experimental/components/input-phone-international/input-phone-international.component.ts

```diff
@@ -0,0 +1,241 @@
+import {TuiChevron} from '@taiga-ui/kit/directives/chevron';
+import {TUI_COUNTRIES, TUI_INTERNATIONAL_SEARCH} from '@taiga-ui/kit/tokens';
+import {tuiGetCallingCode, tuiMaskito} from '@taiga-ui/kit/utils';
+import {TuiCell} from '@taiga-ui/layout/components/cell';
+import type {PolymorpheusContent} from '@taiga-ui/polymorpheus';
+import {validatePhoneNumberLength} from 'libphonenumber-js';
+import type {MetadataJson} from 'libphonenumber-js/core';
+import {filter, from, skip} from 'rxjs';
+
+const NOT_FORM_CONTROL_SYMBOLS = /[^+\d]/g;
+
+@Component({
+    standalone: true,
+    selector: 'input[tuiInputPhoneInternational]',
+    imports: [
+        FormsModule,
+        NgForOf,
+        NgIf,
+        TuiAutoFocus,
+        TuiButton,
+        TuiCell,
+        TuiChevron,
+        TuiDataList,
+        TuiFlagPipe,
+        TuiIcon,
+        TuiTextfield,
+        TuiTextfieldContent,
+        TuiTitle,
+    ],
+    templateUrl: './input-phone-international.template.html',
+    styleUrls: ['./input-phone-international.style.less'],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [
+        tuiAsControl(TuiInputPhoneInternational),
+        tuiFallbackValueProvider(''),
+        tuiAutoFocusOptionsProvider({preventScroll: true}),
+    ],
+    hostDirectives: [MaskitoDirective, TuiWithTextfield, TuiInputMode],
+    host: {
+        '[attr.readonly]': 'readOnly() || null',
+        '[disabled]': 'disabled()',
+        '[value]': 'masked()',
+        '(blur)': 'onTouched()',
+        '(input)': 'onInput()',
+        '(click)': 'open.set(false)',
+        '(beforeinput.capture)': 'onPaste($event)',
+    },
+})
+export class TuiInputPhoneInternational extends TuiControl<string> {
+    @ViewChildren(TuiOption, {read: ElementRef})
+    protected readonly list: QueryList<ElementRef<HTMLButtonElement>> = EMPTY_QUERY;
+
+    protected readonly el = tuiInjectElement<HTMLInputElement>();
+    protected readonly icons = inject(TUI_COMMON_ICONS);
+    protected readonly options = inject(TUI_INPUT_PHONE_INTERNATIONAL_OPTIONS);
+    protected readonly countries = signal(this.options.countries);
+    protected readonly code = signal(this.options.countryIsoCode);
+    protected readonly label = toSignal(inject(TUI_INTERNATIONAL_SEARCH));
+    protected readonly metadata = toSignal(from(this.options.metadata));
+    protected readonly names = toSignal(inject(TUI_COUNTRIES));
+    protected readonly open = tuiDropdownOpen();
+    protected readonly dropdown = tuiDropdown(null);
+    protected readonly search = signal<string>('');
+    protected readonly size = inject(TUI_TEXTFIELD_OPTIONS).size;
+
+    protected readonly mask = tuiMaskito(
+        computed(() => this.computeMask(this.code(), this.metadata())),
+    );
+
+    protected readonly masked = computed(
+        () =>
+            maskitoTransform(this.value(), this.mask() || {mask: /.*/}) || this.el.value,
```

Comment:
Should utility `this.computeMask` return always non-nullable value (for code simplicity) ? 🤔 

```ts
import { MASKITO_DEFAULT_OPTIONS } from '@maskito/core';

// [...]

private computeMask(
    countryIsoCode: TuiCountryIsoCode,
    metadata?: MetadataJson,
): MaskitoOptions {
    if (!metadata) {
        return MASKITO_DEFAULT_OPTIONS;
    }

    // [...]
}
```

```ts
protected readonly mask = tuiMaskito(
    computed(() => this.computeMask(this.code(), this.metadata())),
);

protected readonly masked = computed(
    () => maskitoTransform(this.value(), this.mask()) || this.el.value,
);
```

Reply:
I tried that but seems like something with typings of `tuiMaskito` that allows both signals and values still added `null` to types.

Reply:
Okay

Let's use `MASKITO_DEFAULT_OPTIONS` here

```suggestion
            maskitoTransform(this.value(), this.mask() || MASKITO_DEFAULT_OPTIONS) || this.el.value,
```